### PR TITLE
remove restriction on `fs::Int` in multitaper methods

### LINKED ISSUE #490: `fs::Int=` kwarg restriction seems unnecessary
the `MTConfig` struct accepts any type for the `fs` field (https://github.com/JuliaDSP/DSP.jl/blob/b9c092599ed52224ee6572f1757c8b6a361c6fc7/src/multitaper.jl#L7), but at least one method that (indirectly) creates one (`mt_spectrogram`) restricts `fs` to be an `Int`: https://github.com/JuliaDSP/DSP.jl/blob/master/src/multitaper.jl#L366

### PR DESCRIPTION
**Pull request by kleinschmidt**
Fixes #490 

I went ahead and bumped the patch version; I didn't see any commits that introduced breaking changes between the last release and current master.

### CODE REVIEW

**Comment by palday** [comment]
Note that other functions only restrict `fs` to `Real`, not `Int`:

https://github.com/JuliaDSP/DSP.jl/blob/b9c092599ed52224ee6572f1757c8b6a361c6fc7/src/periodograms.jl#L262-L264

https://github.com/JuliaDSP/DSP.jl/blob/b9c092599ed52224ee6572f1757c8b6a361c6fc7/src/periodograms.jl#L443-L446

https://github.com/JuliaDSP/DSP.jl/blob/f3a246729bd63acac98d85beec24f755e9f21613/src/Filters/design.jl#L238

https://github.com/JuliaDSP/DSP.jl/blob/f3a246729bd63acac98d85beec24f755e9f21613/src/Filters/design.jl#L256

https://github.com/JuliaDSP/DSP.jl/blob/f3a246729bd63acac98d85beec24f755e9f21613/src/Filters/design.jl#L269

https://github.com/JuliaDSP/DSP.jl/blob/f3a246729bd63acac98d85beec24f755e9f21613/src/Filters/design.jl#L283

https://github.com/JuliaDSP/DSP.jl/blob/f3a246729bd63acac98d85beec24f755e9f21613/src/Filters/design.jl#L300

File: src/multitaper.jl [hunk 1]
```diff
@@ -310,7 +310,7 @@ See also [`mt_spectrogram`](@ref).
 mt_spectrogram!
 
 function mt_spectrogram!(output, signal::AbstractVector{T}, n::Int=length(signal) >> 3,
-    n_overlap::Int=n >> 1; fs::Int=1, onesided::Bool=T <: Real,
+    n_overlap::Int=n >> 1; fs=1, onesided::Bool=T <: Real,
```
Comment (hannahilea) [inline src/multitaper.jl:313]:
```suggestion
    n_overlap::Int=n >> 1; fs::Real=1, onesided::Bool=T <: Real,
```
based on 
> Note that other functions only restrict fs to Real, not Int
?

**Comment by kleinschmidt** [comment]
> Note that other functions only restrict fs to Real, not Int:

There's no such restriction in the MTConfig struct itself though, and there are other functions that don't restrict `fs` at all (`mt_coherence`, `mt_cross_power_spectra`, possibly others).

**Comment by kleinschmidt** [comment]
Test failures seem unrelated to the changes here (failing on nighly w/ windows for some unrelated test set).

**Comment by martinholters** [comment]
Fix for the test failures would be at #488.

**Pull request merged**